### ISSUE TITLE
Invalid keyword arguments not flagged as error in TypedDict definition

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6813]
When defining a TypedDict class, additional keyword arguments (including `metadata`) are not allowed, but pyright does not flag this as an error.

```python
# > Specifying a metaclass is not allowed.
class BadTypedDict2(TypedDict, metaclass=type):
    name: str

# This should generate an error because "other" is not an allowed keyword argument.
class BadTypedDict3(TypedDict, other=True):
    name: str
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.343, which I just published. It will also be included in a future release of pylance.
